sort/bogosort.go

```diff
@@ -0,0 +1,32 @@
+// bogosort.go
+// description: Implementation of the bogosort algorithm
+// details:
+// A simple bogosort algorithm implementation. [Wikipedia](https://en.wikipedia.org/wiki/Bogosort)
+// author(s): [Focusucof](https://github.com/Focusucof)
+// see bogosort_test.go for tests
+
+package sort
+
+import (
+	"github.com/TheAlgorithms/Go/constraints"
+	"math/rand"
+	"sort"
+)
+
+// shuffles the given int array
+func shuffle(arr *[]int) {
+	rand.Shuffle(len(*arr), func(i, j int) {
+		(*arr)[i], (*arr)[j] = (*arr)[j], (*arr)[i] // swap values
+	})
```

Comment:
I also think this should be implemented yourself as well rather than depending on the `rand` package but it might be difficult (without the assembly language primitives) so I leave it up to you to decide whether you should do it or not.